src/org/labkey/test/components/html/Table.java

```diff
@@ -79,6 +79,19 @@ public int getRowCount()
         return elementCache().getRows().size();
     }
 
+    /**
+     * For a well formed html table get the text from the th elements in the thead.
+     *
+     * @return A list of the text values contained in the th tags in the thead.
+     */
+    public List<String> getTableHeaderTexts()
+    {
+        List<WebElement> headerEls = Locator.xpath("//thead//th").findElements(this);
```

Comment:
That Locator would find headers in nested tables. We can make it a bit more strict and only find a specific table's header.
```suggestion
        List<WebElement> headerEls = Locator.xpath("./thead/th").findElements(this);
```

Reply:
Changed.